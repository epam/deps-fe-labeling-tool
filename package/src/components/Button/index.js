import { Button, ButtonType } from './Button'
import { IconButton } from './IconButton'
import { TextButton } from './TextButton.styles'

Button.Icon = IconButton

export {
  Button,
  ButtonType,
  TextButton
}
