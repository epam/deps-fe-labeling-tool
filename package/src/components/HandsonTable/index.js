
export * from './models'
export * from './HandsonTable'
