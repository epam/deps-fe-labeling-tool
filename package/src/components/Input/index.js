import { FilterInput } from './FilterInput'
import { Input } from './Input'
import { NumericInput } from './NumericInput'

Input.Numeric = NumericInput
Input.Filter = FilterInput

export {
  Input
}
