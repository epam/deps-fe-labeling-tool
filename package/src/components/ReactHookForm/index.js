export * from './Form'
export * from './FormField'
export * from './FormItem'
export * from './Validators'
export * from './FormFieldType'
