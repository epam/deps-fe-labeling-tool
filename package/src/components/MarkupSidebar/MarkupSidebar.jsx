import React from 'react'
import { Tabs } from '@/components/Tabs'
import { AssignTo } from '@/containers/AssignTo'
import { ObjectProperties } from '@/containers/ObjectProperties'
import { TabsStyled } from './MarkupSidebar.styles'

const { TabPane } = Tabs

const RESOURCE_TAB = {
  PROPERTIES: 'Properties',
  ASSIGN: 'Assign to'
}

const tabsToRender = {
  [RESOURCE_TAB.PROPERTIES]: () => <ObjectProperties />,
  [RESOURCE_TAB.ASSIGN]: () => <AssignTo />
}

const MarkupSidebar = () => (
  <TabsStyled defaultActiveKey={RESOURCE_TAB.PROPERTIES} centered>
    {
      Object.entries(tabsToRender).map(([tabName, renderer]) => (
        <TabPane tab={tabName} key={tabName}>
          {renderer()}
        </TabPane>
      ))
    }
  </TabsStyled>
)

export {
  MarkupSidebar
}
