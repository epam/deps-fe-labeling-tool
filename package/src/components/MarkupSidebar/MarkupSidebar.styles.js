import styled from 'styled-components'
import { Tabs } from '@/components/Tabs'

const TabsStyled = styled(Tabs)`
  padding-bottom: 5rem;

  .ant-tabs-tab {
    margin: 0.8rem 1.8rem;
  }

  .ant-tabs-tab:nth-child(2) {
    margin: 0 2rem;
  }
`

export {
  TabsStyled
}
