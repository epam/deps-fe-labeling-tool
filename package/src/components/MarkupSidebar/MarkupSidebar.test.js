import React from 'react'
import { shallow } from 'enzyme'
import { MarkupSidebar } from '@/components/MarkupSidebar'

describe('Component: LeftSidebar', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<MarkupSidebar />)
  })

  it('should render correct layout', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
