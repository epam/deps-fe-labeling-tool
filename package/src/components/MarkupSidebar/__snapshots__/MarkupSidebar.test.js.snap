// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LeftSidebar should render correct layout 1`] = `
<MarkupSidebarstyles__TabsStyled
  centered={true}
  defaultActiveKey="Properties"
>
  <TabPane
    key="Properties"
    tab="Properties"
  >
    <Connect(ObjectProperties) />
  </TabPane>
  <TabPane
    key="Assign to"
    tab="Assign to"
  >
    <Connect(AssignTo) />
  </TabPane>
</MarkupSidebarstyles__TabsStyled>
`;
